service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java

```diff
@@ -180,6 +180,11 @@ public String getRedirectToTppUriFromPaymentRepo(String externalId) {
     TppInfoEntity tppInfo = commonPaymentData.get().getAuthorizations().get(0).getPaymentData()
         .getTppInfo();
 
+    ScaStatus scaStatus = commonPaymentData.get().getAuthorizations().get(0).getScaStatus();
+
+    if (scaStatus.equals(ScaStatus.FAILED)) {
+      return tppInfo.getNokRedirectUri();
```

Comment:
Since both redirect URI params are optional (AFAIK): what should we do if there is only a `RedirectUri` set but no `NokRedirectUri`? Fall back to `RedirectUri`? What if there is not `RedirectUri` at all?

Reply:
The redirectUri is requried for the redirect sca approach. When the tpp does not pass a nokRedirectUri to the aspsp, the users will be redirected to its redirectUri on sca failure.